### PR TITLE
feat(project): add items bindable

### PR DESCRIPTION
**Pull request by bas080**
This enables users to use and array with values to auto-complete on. When using
this bindable it disables the performing of requests.

Contains the necessary documentation which explains what methods can be
overwritten and the newly defined items bindable. Also updated the altered
bindables behavior documentation

### CODE REVIEW

File: src/component/autocomplete.js [hunk 1]
```diff
@@ -74,6 +72,9 @@ export class AutoCompleteCustomElement {
       return logger.error('auto complete requires resource or items bindable to be defined');
     }
 
+    this.search = this.label(this.value);
+    this.justSelected = true;
```
Comment (RWOverdijk) [inline src/component/autocomplete.js:76]: Sloppy code style

File: src/component/autocomplete.js [hunk 2]
```diff
@@ -131,14 +132,21 @@ export class AutoCompleteCustomElement {
   }
 
   /**
-   * returns HTML that wraps matching substrings with strong tags
+   * returns HTML that wraps matching substrings with strong tags. If not
```
Comment (RWOverdijk) [inline src/component/autocomplete.js]: Cut on new sentence or append on the end.

File: src/component/autocomplete.js [hunk 3]
```diff
@@ -148,16 +156,11 @@ export class AutoCompleteCustomElement {
    */
   attached() {
     this.inputElement    = this.element.querySelectorAll('input')[0];
-    this.dropdownElement = this.element.querySelectorAll('#dropdown')[0];
-
-    let openDropdown = (show = true) => {
-      this.showResults = show;
-    };
+    this.dropdownElement = this.element.querySelectorAll('.dropdown.open')[0];
```
Comment (RWOverdijk) [inline src/component/autocomplete.js:159]: Why is this still needed?

**Review by RWOverdijk — CHANGES_REQUESTED** [review]

File: src/component/autocomplete.js (hunk 3 above)
Comment (bas080) [reply src/component/autocomplete.js:159]: Because bootstrap.js will add and remove the open class for you

File: src/component/autocomplete.js [hunk 4]
```diff
@@ -299,10 +334,13 @@ export class AutoCompleteCustomElement {
...
+    // only assign limit to query if it is defined. Allows to default to server
+    // limit when limit bindable is set to falsy value
     if (this.limit) {
       query.limit = this.limit;
     }
 
     return query;
   }
 }
+
```
Comment (RWOverdijk) [inline src/component/autocomplete.js]: Remove newline

File: src/component/autocomplete.js [hunk 5]
```diff
@@ -211,49 +217,78 @@ export class AutoCompleteCustomElement {
...
+    // when resource is not defined it will not perform a request. Instead it
+    // will search for the first items that pass the predicate
+    if (this.items) {
+      this.results = this.sort(this.filter(this.items));
+
+      return Promise.resolve();
     }
 
-    return this.findResults(this.searchQuery(this.search)).then(results => {
+    let promise = this.findResults(this.searchQuery(this.search)).then(results => {
```
Comment (RWOverdijk) [inline src/component/autocomplete.js]: Why not use: `this.lastFindPromise = ....`?

It's a sync process as far as I can see, so the assignment is basically the same, isn't it?

**Review by RWOverdijk — CHANGES_REQUESTED** [review]

File: src/component/autocomplete.js (hunk 5 above)
Comment (bas080) [reply src/component/autocomplete.js]: that's correct

**Comment by RWOverdijk** [comment]
@VMBindraban and @jeremyvergnas please review.

File: src/component/autocomplete.js [hunk 6]
```diff
@@ -211,49 +217,76 @@ export class AutoCompleteCustomElement {
...
+    // when resource is not defined it will not perform a request. Instead it
+    // will search for the first items that pass the predicate
+    if (this.items) {
+      this.results = this.sort(this.filter(this.items));
+
+      return Promise.resolve();
+    }
+
+    this.lastFindPromise = this.findResults(this.searchQuery(this.search)).then(results => {
+      if (this.lastFindPromise !== promise) {
```
Comment (jeremyvergnas) [inline src/component/autocomplete.js:238]: Can be removed I guess

**Pull request merged**